src/components/swap/SwapModalFooter.tsx

```diff
@@ -121,46 +175,135 @@ export default function SwapModalFooter({
   swapQuoteReceivedDate: Date | undefined
   fiatValueInput: { data?: number; isLoading: boolean }
   fiatValueOutput: { data?: number; isLoading: boolean }
+  shouldLogModalCloseEvent: boolean
+  setShouldLogModalCloseEvent: (shouldLog: boolean) => void
+  showAcceptChanges: boolean
+  onAcceptChanges: () => void
 }) {
   const transactionDeadlineSecondsSinceEpoch = useTransactionDeadline()?.toNumber() // in seconds since epoch
   const isAutoSlippage = useUserSlippageTolerance()[0] === 'auto'
   const [routerPreference] = useRouterPreference()
-  const routes = getRoutingDiagramEntries(trade)
+  const routes = trade ? getRoutingDiagramEntries(trade) : []
+  const [lastExecutionPrice, setLastExecutionPrice] = useState(trade?.executionPrice)
+  const [priceUpdate, setPriceUpdate] = useState<number>()
+  const theme = useTheme()
+  const { chainId } = useWeb3React()
+  const nativeCurrency = useNativeCurrency(chainId)
+
+  useEffect(() => {
+    if (trade && lastExecutionPrice && !trade?.executionPrice.equalTo(lastExecutionPrice)) {
+      setPriceUpdate(getPriceUpdateBasisPoints(lastExecutionPrice, trade?.executionPrice))
+      setLastExecutionPrice(trade?.executionPrice)
+    }
+  }, [lastExecutionPrice, setLastExecutionPrice, trade])
+
+  useEffect(() => {
+    if (trade && shouldLogModalCloseEvent && showAcceptChanges) {
+      sendAnalyticsEvent(
+        SwapEventName.SWAP_PRICE_UPDATE_ACKNOWLEDGED,
+        formatSwapPriceUpdatedEventProperties(trade, priceUpdate, SwapPriceUpdateUserResponse.REJECTED)
+      )
+    }
+    setShouldLogModalCloseEvent(false)
+  }, [shouldLogModalCloseEvent, showAcceptChanges, setShouldLogModalCloseEvent, trade, priceUpdate])
+
+  const details = useMemo(() => {
```

Comment:
I think we should remove this and generally avoid defining content of components as arrays as long as they're not dynamic / coming from the API.

Instead, we should explicitly define each `detail` in JSX, such as:
```
<SwapModalDetailRow label={""} value={""} />
```
and so on.

That gives us more control and flexibility as well as it's more light-weight and React friendly.